Explorer/Assets/DCL/WebRequests/Analytics/WebRequestsContainer.cs

```diff
@@ -1,24 +1,57 @@
 using DCL.Web3.Identities;
 using DCL.WebRequests.Analytics.Metrics;
+using System;
+using UnityEngine;
 
 namespace DCL.WebRequests.Analytics
 {
     public class WebRequestsContainer
     {
-        public IWebRequestController WebRequestController { get; private set; }
+        public IWebRequestController WebRequestController { get; }
 
-        public IWebRequestsAnalyticsContainer AnalyticsContainer { get; private set; }
+        public IWebRequestsAnalyticsContainer AnalyticsContainer { get; }
 
-        public static WebRequestsContainer Create(IWeb3IdentityCache web3IdentityProvider)
+        private WebRequestsContainer(IWebRequestController webRequestController, IWebRequestsAnalyticsContainer analyticsContainer)
         {
-            var analyticsContainer = new WebRequestsAnalyticsContainer().AddTrackedMetric<ActiveCounter>()
-                                                                        .AddTrackedMetric<Total>()
-                                                                        .AddTrackedMetric<TotalFailed>()
-                                                                        .AddTrackedMetric<BandwidthDown>()
-                                                                        .AddTrackedMetric<BandwidthUp>();
-
-            var webRequestController = new LogWebRequestController(new WebRequestController(analyticsContainer, web3IdentityProvider));
-            return new WebRequestsContainer { WebRequestController = webRequestController, AnalyticsContainer = analyticsContainer };
+            WebRequestController = webRequestController;
+            AnalyticsContainer = analyticsContainer;
         }
+
+        public static WebRequestsContainer Create(IWeb3IdentityCache web3IdentityProvider, WebRequestsContainerParams containerParams)
+        {
+            var analyticsContainer = new WebRequestsAnalyticsContainer()
+                                    .AddTrackedMetric<ActiveCounter>()
+                                    .AddTrackedMetric<Total>()
+                                    .AddTrackedMetric<TotalFailed>()
+                                    .AddTrackedMetric<BandwidthDown>()
+                                    .AddTrackedMetric<BandwidthUp>();
+
+            IWebRequestController webRequestController = new LogWebRequestController(
+                new WebRequestController(analyticsContainer, web3IdentityProvider)
+            );
+
+            if (containerParams.ShouldApplyDelay())
+            {
+                ReportHub.Log(ReportCategory.WEB_REQUESTS, "Artificial delay enabled");
+                webRequestController = new ArtificialDelayWebRequestController(webRequestController, containerParams.ArtificialDelaySeconds);
+            }
+
+            return new WebRequestsContainer(webRequestController, analyticsContainer);
+        }
+    }
+
+    [Serializable]
+    public class WebRequestsContainerParams
+    {
+        [SerializeField] private bool useDelay = true;
```

Comment:
Why are you defaulting this to true?

Reply:
good mention, I put it as false by default in the plugin settings